tests/PHPStan/Analyser/data/bug-1396.php

```diff
@@ -0,0 +1,37 @@
+<?php declare(strict_types = 1);
+
+namespace Bug1396;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\ConstantScalarType;
+use function PHPStan\Testing\assertType;
+
+class TypeSpecifier2
+{
+	/**
+	 * @api
+	 */
+	public function specifyTypesInCondition(
+		ConstantScalarType $constantType
+	): SpecifiedTypes
+	{
+		if ($constantType->getValue() === null) {
```

Comment:
it seems this nullcheck is somehow messing it up 

maybe @rvanvelzen  has an idea?

Reply:
That's interesting. Null isn't even a scalar, right? Is this ever triggered and is there even a ConstantType that implements returning null via getValue?

Reply:
Ah one more thing came to mind: could it be related to the fact that phpstan remembers what the method returned? Maybe this needs to he invalidated below in the if scope

Reply:
@rajyan do you have an idea on how to approach this problem? maybe you even have time for a fix?